### PR TITLE
Change number_to_currency behavior for checking negativity of a number

### LINKED ISSUE #23320: number_to_currency should not return negative zero
```
[17] pry(main)> helper.number_to_currency BigDecimal('0')*-1
=> "$-0.00"
```

negative zero is not a humanized money amount, so it should never be returned by this function.

### PR DESCRIPTION
**Pull request by prathamesh-sonpatki**
r? @rafaelfranca 
Fixes #23320

### CODE REVIEW

File: activesupport/lib/active_support/number_helper/number_to_currency_converter.rb [hunk 1]
```diff
@@ -22,6 +24,10 @@ def is_negative?(number)
           number.to_f.phase != 0
```
Comment (rafaelfranca) [inline activesupport/lib/active_support/number_helper/number_to_currency_converter.rb]: We can just use `negative?` here and avoid the gsub and the new method.

**Comment by pcai** [comment]
Thank you for submitting. I was just looking at this myself and wondering - wouldn't it make more sense to fix [is_negative?](https://github.com/rails/rails/blob/master/activesupport/lib/active_support/number_helper/number_to_currency_converter.rb#L22) instead of adding a new function? It does not seem intuitive for us to declare a number to be both zero and negative.

File: activesupport/lib/active_support/number_helper/number_to_currency_converter.rb [hunk 2]
```diff
@@ -9,6 +9,8 @@ def convert
 
         if is_negative?(number)
```
Comment (rafaelfranca) [inline activesupport/lib/active_support/number_helper/number_to_currency_converter.rb]: `if number.to_f.negative?`

Comment (prathamesh-sonpatki) [inline activesupport/lib/active_support/number_helper/number_to_currency_converter.rb]: If we do that then following test fails - 

```
assert_equal("(0.00)", number_helper.number_to_currency(-0.0, {:unit => "", :negative_format => "(%n)"}))
```

As `negative_format` did not get assigned to `format`

File: activesupport/lib/active_support/number_helper/number_to_currency_converter.rb (hunk 1 above)
Comment (prathamesh-sonpatki) [inline activesupport/lib/active_support/number_helper/number_to_currency_converter.rb]: I think `phase` is required for specific case of `-0.0`

``` ruby
>> -0.0.to_f.phase
=> 3.141592653589793
>> -0.0.to_f.negative?
=> false
>> 
```

**Comment by rafaelfranca** [comment]
O_O seems this behavior is intentional https://github.com/rails/rails/pull/6512

**Comment by prathamesh-sonpatki** [comment]
@rafaelfranca Yup :) But only case missed was with default `negative format` 

like `number_helper.number_to_currency(-0.0, { :unit => ""}) => -0.0`

**Comment by rafaelfranca** [comment]
I don't know if any currency in the world treat -0 as a different thing as +0. I'll ask more opinions.

cc @pixeltrix @matthewd @chancancode @senny

**Comment by rafaelfranca** [comment]
@prathamesh-sonpatki the default negative format should not make any change here, if -0 is considered negative so it should be -0.0 with the default format.

**Comment by prathamesh-sonpatki** [comment]
True,  currently it does not
```
>> helper.number_to_currency(-0.0, { :unit => ""})
=> "-0.00"
>> helper.number_to_currency(-0, { :unit => ""})
=> "0.00"
>> 
```

**Comment by prathamesh-sonpatki** [comment]
@rafaelfranca Made changes as per our discussion

**Pull request merged**